docs/tutorials/ros2.md

```diff
@@ -0,0 +1,151 @@
+platforms = ["linux-64"]
+
+[tasks]
+
+[dependencies]
+```
+The `channels` you added to the `init` command are repositories of packages, you can search in  these repositories through our [prefix.dev](https://prefix.dev/channels) website.
+The `platforms` are the systems you want to support, in pixi you can support multiple platforms, but you have to define which platforms, so pixi can test if those are supported for your dependencies.
+For the rest of the fields, you can fill them in as you see fit.
+
+## Add ROS2 dependencies
+To use a pixi project you don't need any dependencies on your system, all the dependencies you need should be added through pixi, so other users can use your project without any issues.
+
+Let's start with the `turtlesim` example
+```shell
+pixi add ros-humble-desktop ros-humble-turtlesim
+```
+This will add the `ros-humble-desktop` and `ros-humble-turtlesim` packages to your manifest.
+Depending on your internet speed this might take a minute, as it will also install ROS in your project folder (`.pixi`).
+
+Now run the `turltesim` example
+```shell
+pixi run ros2 run turtlesim turtlesim_node
+```
+Congratulations you have ROS2 running on your machine with pixi!
+
+??? example Some more fun with the turtle
+    To control the turtle you can run the following command in a new terminal
+    ```shell
+    cd my_ros2_project
+    pixi run ros2 run turtlesim turtle_teleop_key
+    ```
+    Now you can control the turtle with the arrow keys on your keyboard.
+    ![Turtlesim control](https://private-user-images.githubusercontent.com/12893423/319520632-c80c5054-3335-49c7-9671-92bd8702708b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIyMjI2MDQsIm5iZiI6MTcxMjIyMjMwNCwicGF0aCI6Ii8xMjg5MzQyMy8zMTk1MjA2MzItYzgwYzUwNTQtMzMzNS00OWM3LTk2NzEtOTJiZDg3MDI3MDhiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA0VDA5MTgyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1MjU0ZGY0ZjQ1OTg4ZjYyNTE1Y2I2MmYwNWU5YzQwYzFlNGExMzkyNDhjOGY4N2I4NDQ5ZTA3MzI4MzhiNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4dD3FdMZ7VBRPTLcnSPlWsSWkx4pgjKq24NEL-TnW0g)
+
+
+## Add a custom Python node
+As ros works with custom nodes, let's add a custom node to our project.
+```shell
+pixi run ros2 pkg create --build-type ament_python --destination-directory src --node-name my_node my_package
+```
+
+To build the package we need some more dependencies:
+```shell
+pixi add colcon-core colcon-ros
+```
+Add the created initialization script for the ros workspace to your manifest file.
+
+Then run the build command
+```shell
+pixi run colcon build
+```
+
+This will create a sourcable script in the `install` folder, you can source this script through an activation script to use your custom node.
```

Comment:
```suggestion
This will create a sourceable script in the `install` folder, you can source this script through an activation script to use your custom node.
```